job/jobcache.go

```diff
@@ -1,36 +1,86 @@
 package job
 
 import (
+	"os"
+	"os/signal"
 	"sync"
+	"time"
 )
 
-var (
-	AllJobs = &JobCache{
-		Jobs:   make(map[string]*Job),
-		rwLock: sync.Mutex{},
+type JobCache interface {
+	Get(id string) *Job
+	GetAll() map[string]*Job
+	Set(j *Job)
+	Delete(id string)
+	Persist() error
+}
+
+type MemoryJobCache struct {
+	Jobs            map[string]*Job
+	rwLock          sync.Mutex
```

Comment:
@ajvb If it's really a read-write lock, use `sync.RWMutex` so that readers can call `.RLock()` / `.RUnlock()`